rvc-implementation.go

```diff
@@ -0,0 +1,98 @@
+// Copyright (c) 2017 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Description: The true virtcontainers implementation of the RVC interface.
+// This indirection is required to allow an alternative implemenation to be
+// used for testing purposes.
+
+package main
+
+import (
+	"syscall"
+
+	"github.com/Sirupsen/logrus"
+	vc "github.com/containers/virtcontainers"
+)
+
+// virtcontainers implementation type
+type vcImpl struct {
+}
+
+func (impl *vcImpl) SetLogger(logger logrus.FieldLogger) {
+	vc.SetLogger(logger)
```

Comment:
@jodh-intel I agree with sboeuf's comment below, why not have the mock implementation in virtcontainers package?  That way we do not need to introduce the indirections which add yet another call and also make it a little less intuitive with code calling into vc for the state constants and vci for the actual lifecycle calls.